omod-common/src/main/java/org/openmrs/module/webservices/rest/web/api/impl/RestServiceImpl.java

```diff
@@ -632,16 +632,23 @@ private void eliminateCandidateSearchHandlersWithMissingRequiredParameters(Set<S
 		return resourceHandlers;
 	}
 	
-	/* (non-Javadoc)
-	 * @see org.openmrs.module.webservices.rest.web.api.RestService#getSearchHandlers(java.lang.String)
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.api.RestService#getAllSearchHandlers()
+	 * @should return all search handlers if search handlers have been initialized
```

Comment:
Sure, the interface service method signature definitely defines the contract. But it is debatable if some/all/none/which of the @shoulds are part of the service/API contract. Are you only talking about the @shoulds here or in general?

Because for ex. you might not want to force a different impl of the service to always return the Resource with the lowest order if multiple Resources are found for the same class. This allows different impls for the same service.

If you only mean the schoulds here I can move them over to the service interface.